### ISSUE TITLE
Nuxt does not detect when port is already used by other express app

### ISSUE THREAD

**Issue by julian-eckhardt** [nuxt/nuxt.js#6878]
### Version

[v2.11.0](https://github.com/nuxt.js/releases/tag/v2.11.0)

### Reproduction link

[https://github.com/julian-eckhardt/nuxt-express-port-confusion](https://github.com/julian-eckhardt/nuxt-express-port-confusion)

### Steps to reproduce

- checkout repository
- run `npm install` in both app directories
- cd into `express-app` and run `npm run start`
- open second terminal, cd into `nuxt-app` and run `npm run dev`

### What is expected ?

Nuxt reports successful startup of server, so a normal behaviour (i.e. the demo nuxt app showing up) is expected. On a technical level, nuxt should choose a random port under these circumstances and report it to the user.

### What is actually happening?

Nuxt reports successful startup of server, but cannot bind to the port. The user accesses the port and gets served the content provided by our second express server instead. This behaviour is very unfortunate as the bare `Cannot GET /` error message provided by our express app if it does not contain an endpoint for `GET /` is easily misunderstood as a nuxt error.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10208">#c10208</a>)</em></sub></div>

**Comment by aldarund** [comment]
It should like your described, if port is taken it will use another free port and report to user.

if that dont happen to you possible cause is that your nuxt and express app binds to different interface e.g. localhost/127.0.0.1 and your network adapter ip for example

**Comment by talebi1** [comment]
use --port 0 to assign a randomly available port to your nuxt application (https://nuxtjs.org/faq/host-port/)

**Comment by julian-eckhardt** [comment]
@talebi1 thanks - I circumvented this by changing the port my express app is running on, which works just as fine for my use case.

@aldarund I'm slowly understanding the root cause for this… It seems to be the case that nodejs itself tries to bind to `::` if available and to `0.0.0.0` otherwise. As the operating system (Mac OS in my case) usually makes the IPv6 version also listen on the IPv4 interface ([Node.JS Docs on this topic](https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback)), both interfaces are used, but as `localhost` resolves to either one of them, the other one will simply not be accessed if you rely on the hostname.

I think we can close this, as it is more of a node or operating system issue than something related to nuxt.  At least I raised some awareness about the issue.
